src/segger/data/io.py

```diff
@@ -253,12 +254,48 @@ def load_boundaries(
         ]
 
         # Load the dataset lazily with filters applied for the bounding box
-        boundaries_df = dd.read_parquet(path, columns=columns_to_read, filters=filters)
+        columns = set(dd.read_parquet(path).columns)
```

Comment:
Three comments on this code, which is converting the `GeoDataFrame` data into a Dask `DataFrame`.
1) this code is in principle redundant because when constructing the `pyg` data, the Dask `DataFrame` is turned into a `GeoDataFrame`. So a refactoring to just accept a `GeoDataFrame` would be better.
2) this code is not efficient as it is now, but addressing 1 is the way to go, not making this code efficient
3) biggest point. As I explain in the TODO, I am not sure that the "bbox covering metadata" ([see `geopandas` docs](https://geopandas.org/en/stable/docs/reference/api/geopandas.read_parquet.html)) are actually saved by `SpatialData.write()`. So if this is slow even after 1 is addressed, it is probably because of this. Easy to fix.